software/grapherrific.livecodescript

```diff
@@ -0,0 +1,1477 @@
+      repeat for each line tPtr in tPointers
+         put item 1 of tPtr into tPtr
+         if tNode["type"] is among the keys of pMain[pDocKey][tPtr] then
+            put tPtr into tNode["Y"]
+            put item 1 of line 1 of pMain[pDocKey][tPtr][tNode["type"]] into tAlmost
+            put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["X"]
+            put item 1 of line 2 of pMain[pDocKey][tPtr][tNode["type"]] into tAlmost
+            put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["Z"]
+            -- Xcoord
+            -- Zcoord       
+         end if
+      end repeat
+   end if
+   -- directed links are more complicated
+   switch tNode["type"]
+      case "flow"
+         repeat for each line tPtr in tPointers
+            put item 1 of tPtr into tPtr
+            if "prev" is among the keys of pMain[pDocKey][tPtr] then 
+               put item 1 of pMain[pDocKey][tPtr]["prev"] into tAlmost
+               put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["X"]
+            end if
+            if "next" is among the keys of pMain[pDocKey][tPtr] then 
+               put item 1 of pMain[pDocKey][tPtr]["next"] into tAlmost
+               put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["Z"]
+            end if
+         end repeat
+         break
+      case "dive"
+         repeat for each line tPtr in tPointers
+            put item 1 of tPtr into tPtr
+            if "skim" is among the keys of pMain[pDocKey][tPtr] then 
+               put item 1 of pMain[pDocKey][tPtr]["skim"] into tAlmost
+               put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["X"]
+            end if
+            if "read" is among the keys of pMain[pDocKey][tPtr] then 
+               put item 1 of pMain[pDocKey][tPtr]["read"] into tAlmost
+               put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["Z"]
+            end if
+         end repeat
+         break
+      case "rise"
+         repeat for each line tPtr in tPointers
+            put item 1 of tPtr into tPtr
+            if "bulk" is among the keys of pMain[pDocKey][tPtr] then 
+               put item 1 of pMain[pDocKey][tPtr]["bulk"] into tAlmost
+               put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["X"]
+            end if
+            if "trim" is among the keys of pMain[pDocKey][tPtr] then 
+               put item 1 of pMain[pDocKey][tPtr]["trim"] into tAlmost
+               put item 1 of pMain[pDocKey][tAlmost]["myStep"] into tNode["Z"]
+            end if
+         end repeat
+         break
+   end switch
+   -- tNode is either false
+   -- or it's an array with type, X, Y, Z
+   return tNode
+end isAlink
+
+function areLinked pX,pZ
+   put "function areLinked" into xX["write"];    devLog xX
+   -- pX & pZ are each a single record number
+   -- if there is already a link connecting them, return true
+   -- otherwise return false
+   put false into tLink
+   put listOfLinks(pX) into pXlinks
```

Comment:
@matthewmaier Use `t` or `the` as prefix for temporary variables. A more efficient way to do this is:

````
put listOfLinks(pX) into tXlinks
put listOfLinks(pY) into tYlinks
split tXlinks with comma as set
split tZlinks with comma as set
intersect tXlinks with tZlinks
return the number of elements of tXlinks > 0
````